js/src/hooks/useGoogleAdsAccountReady.js

```diff
@@ -4,18 +4,22 @@
 import useGoogleAdsAccount from '.~/hooks/useGoogleAdsAccount';
 import useGoogleAdsAccountStatus from '.~/hooks/useGoogleAdsAccountStatus';
 
+/**
+ * Hook to check if the Google Ads account is ready.
+ * This is used to determine if the user can proceed to the next step.
+ *
+ * @return {boolean|null} Whether the Google Ads account is ready. `null` if the state is not yet determined.
+ */
 const useGoogleAdsAccountReady = () => {
 	const { hasGoogleAdsConnection } = useGoogleAdsAccount();
 	const { hasAccess, step } = useGoogleAdsAccountStatus();
 
-	// Ads is ready when we have a connection and verified and verified access.
-	// Billing is not required, and the 'link_merchant' step will be resolved
-	// when the MC the account is connected.
-	return (
+	const isReady =
 		hasGoogleAdsConnection &&
 		hasAccess &&
-		[ '', 'billing', 'link_merchant' ].includes( step )
-	);
+		[ '', 'billing', 'link_merchant' ].includes( step );
+
+	return isReady !== null ? isReady : null;
```

Comment:
The ternary here seems redundant because it means:

If `isReady` is !== `null`, return `isReady`
If `isReady` is === `null`, return `null`

This can be simplified to:

```suggestion
	return isReady;
```

However, it seems this all be simplified to a boolean return value by wrapping the expression in parenthesis like this (and updating the docblock):

```js
	const isReady = (
		hasGoogleAdsConnection &&
		hasAccess &&
		[ '', 'billing', 'link_merchant' ].includes( step )
	);
```

Reply:
@ankitrox I see you updated this slightly in [7f20af7](https://github.com/woocommerce/google-listings-and-ads/pull/2618/commits/7f20af762939b5a603d60815bc4b614ae8ac656c) but still think the suggestions here would be better, unless there is some reason to retain the `null` return value?

Reply:
@joemcgill This has been addressed here: https://github.com/woocommerce/google-listings-and-ads/blob/9137ef6b60a80e210080689ae679946ea1078126/js/src/hooks/useGoogleAdsAccountReady.js#L24-L26

Reply:
Also: https://github.com/woocommerce/google-listings-and-ads/blob/9137ef6b60a80e210080689ae679946ea1078126/js/src/hooks/useGoogleAdsAccountReady.js#L28-L32